packages/intrepid2/src/Projection/Intrepid2_LagrangianInterpolationDef.hpp

```diff
@@ -185,7 +188,8 @@ LagrangianInterpolation<SpT>::getDofCoordsAndCoeffs(
       }
     }
 
-    auto edgeScale = isBasisTriOrTet ? 2.0 :1.0;
+    //depending on how the reference basis is defined, the edges are scaled differently
+    auto edgeScale = (isBasisTriOrTet||isBasisI1) ? 2.0 :1.0;
```

Comment:
The logic here is if(triangle or tet or first order hex or first order tet) edgeScale = 2. Why is that?

Reply:
It really depends on how the basis are defined. We should uniform that at some point but we would break backward compatibility if somebody uses the bases in a weird way.
The convention used to build the first order basis are different the the ones used for high order basis, so, e.g. HGRAD_HEX_C1 is different then HGRAD_HEX_Cn with degree=1.

As for the difference between TRI/TET and QUAD/HEX, it also depends on the fact that the "reference" LINE elemet goes from -1 to 1, and all the edges in the reference HEX/QUAD do the same, however the TET/TRI are built considering a reference edge of size 1, which is were the factor 2 arises..

Reply:
B.t.w. This is partially the reason why @kyungjoo-kim does not like to use interpolations. In fact they heavily depend on the conventions  used in the bases and orientations. But they are fast. Alternatively you could use the projections (e.g. the l2 projections):
https://github.com/trilinos/Trilinos/blob/master/packages/intrepid2/src/Projection/Intrepid2_ProjectionToolsDefL2.hpp